cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -111,20 +123,156 @@ void batched_decompress(compression_type compression,
+                                                      nvcomp_args.output_data_sizes.data(),
                                                       actual_uncompressed_data_sizes.data(),
                                                       num_chunks,
                                                       scratch.data(),
                                                       scratch.size(),
-                                                      nvcomp_args.uncompressed_data_ptrs.data(),
+                                                      nvcomp_args.output_data_ptrs.data(),
                                                       nvcomp_statuses.data(),
                                                       stream.value());
   CUDF_EXPECTS(nvcomp_status == nvcompStatus_t::nvcompSuccess, "unable to perform decompression");
 
   convert_status(nvcomp_statuses, actual_uncompressed_data_sizes, statuses, stream);
 }
+
+// Dispatcher for nvcompBatched<format>CompressGetTempSize
+auto batched_compress_temp_size(compression_type compression,
+                                size_t batch_size,
+                                size_t max_uncompressed_chunk_bytes)
+{
+  size_t temp_size             = 0;
+  nvcompStatus_t nvcomp_status = nvcompStatus_t::nvcompSuccess;
+  switch (compression) {
+    case compression_type::SNAPPY:
+      nvcomp_status = nvcompBatchedSnappyCompressGetTempSize(
+        batch_size, max_uncompressed_chunk_bytes, nvcompBatchedSnappyDefaultOpts, &temp_size);
+      break;
+
+    case compression_type::DEFLATE:
+#if NVCOMP_HAS_DEFLATE
+      nvcomp_status = nvcompBatchedDeflateCompressGetTempSize(
+        batch_size, max_uncompressed_chunk_bytes, nvcompBatchedDeflateDefaultOpts, &temp_size);
+      break;
+#else
+      CUDF_FAIL("Unsupported compression type");
+#endif
+    default: CUDF_FAIL("Unsupported compression type");
+  }
+
+  CUDF_EXPECTS(nvcomp_status == nvcompStatus_t::nvcompSuccess,
+               "Unable to get scratch size for compression");
+  return temp_size;
+}
+
+// Dispatcher for nvcompBatched<format>CompressGetMaxOutputChunkSize
+size_t batched_compress_get_max_output_chunk_size(compression_type compression,
+                                                  uint32_t max_uncompressed_chunk_bytes)
+{
+  size_t max_comp_chunk_size = 0;
+  nvcompStatus_t status      = nvcompStatus_t::nvcompSuccess;
+  switch (compression) {
+    case compression_type::SNAPPY:
+      status = nvcompBatchedSnappyCompressGetMaxOutputChunkSize(
+        max_uncompressed_chunk_bytes, nvcompBatchedSnappyDefaultOpts, &max_comp_chunk_size);
+      break;
+    case compression_type::DEFLATE:
+#if NVCOMP_HAS_DEFLATE
+      status = nvcompBatchedDeflateCompressGetMaxOutputChunkSize(
+        max_uncompressed_chunk_bytes, nvcompBatchedDeflateDefaultOpts, &max_comp_chunk_size);
+      break;
+#else
+      CUDF_FAIL("Unsupported compression type");
+#endif
+    case compression_type::ZSTD: [[fallthrough]];
```

Comment:
This line with `fallthrough` is not present in other switch cases in this PR. would it make sense to add to others too?